app/assets/javascripts/browse-columns.js

```diff
@@ -1,395 +1,323 @@
+      var sectionState = this.parsePathname(sectionPathname)
+      this.getSectionData(sectionState, true)
+      this.loadSectionFromState(sectionState, true)
     }
+    this.loadSectionFromState(state, true)
   }
 
-  GOVUK.BrowseColumns = BrowseColumns
+  BrowseColumns.prototype.loadSectionFromState = function (state, poppingState) {
+    if (state.subsection) {
+      this.showSubsection(state)
+    } else {
+      this.showSection(state)
+    }
+
+    if (typeof poppingState === 'undefined') {
+      history.pushState(state, '', state.path)
+      this.trackPageview(state)
+    }
+  }
+
+  BrowseColumns.prototype.showRoot = function () {
+    this.$section.innerHTML = ''
+    this.displayState = 'root'
+  }
+
+  BrowseColumns.prototype.showSection = function (state) {
+    this.setContentIDMetaTag(state.sectionData.content_id)
+    this.setNavigationPageTypeMetaTag(state.sectionData.navigation_page_type)
+    state.title = this.getTitle(state.slug)
+    this.setTitle(state.title)
+    this.$section.innerHTML = state.sectionData.html
+
+    this.highlightSection('root', state.path)
+    this.updateBreadcrumbs(state)
+
+    if (this.displayState === 'subsection') {
+      this.changeColumnVisibility(1)
+    } else {
+      this.changeColumnVisibility(2)
+    }
+    this.$section.querySelector('.js-heading').focus()
+  }
+
+  BrowseColumns.prototype.changeColumnVisibility = function (columns) {
+    if (columns === 3) {
+      this.$module.classList.remove('browse--two-columns')
+      this.$module.classList.add('browse--three-columns')
+    } else if (columns === 2) {
+      this.$module.classList.add('browse--two-columns')
+      this.$module.classList.remove('browse--three-columns')
+    } else {
+      this.$module.classList.remove('browse--two-columns')
+      this.$module.classList.remove('browse--three-columns')
+    }
+  }
+
+  BrowseColumns.prototype.showSubsection = function (state) {
+    this.setContentIDMetaTag(state.sectionData.content_id)
+    this.setNavigationPageTypeMetaTag(state.sectionData.navigation_page_type)
+    state.title = this.getTitle(state.slug)
+    this.setTitle(state.title)
+    this.$subsection.innerHTML = state.sectionData.html
+    this.highlightSection('section', state.path)
+    this.highlightSection('root', '/browse/' + state.section)
+    this.updateBreadcrumbs(state)
+
+    this.changeColumnVisibility(3)
+    this.$subsection.querySelector('.js-heading').focus()
+  }
+
+  BrowseColumns.prototype.getTitle = function (slug) {
+    var $link = this.$module.querySelector('a[href$="/browse/' + slug + '"]')
+    var $heading = $link.querySelector('h3')
+    if ($heading) {
+      return $heading.textContent
+    } else {
+      return $link.textContent
+    }
+  }
+
+  BrowseColumns.prototype.setTitle = function (title) {
+    document.title = title + ' - GOV.UK'
+  }
+
+  BrowseColumns.prototype.setContentIDMetaTag = function (contentId) {
```

Comment:
It'd be more idiomatic to our regular naming conventions to call this `setContentIdMetaTag`

Reply:
Changed, thanks.